beep/generate_protocol.py

```diff
@@ -528,84 +497,84 @@ def generate_procedure_regcyclev3(self, protocol_index, proc_dict, reg_param):
+        obj.insert_discharge_regcyclev2(reg_discharge_idx, reg_param)
 
         # Start of main loop of regular cycles
         reg_charge_idx = 59 + 1
-        proc_dict = self.insert_charge_regcyclev3(proc_dict, reg_charge_idx, reg_param)
+        obj.insert_charge_regcyclev3(reg_charge_idx, reg_param)
         reg_discharge_idx = 59 + 5
-        proc_dict = self.insert_discharge_regcyclev2(proc_dict, reg_discharge_idx, reg_param)
+        obj.insert_discharge_regcyclev2(reg_discharge_idx, reg_param)
 
         # Storage cycle
         reg_storage_idx = 93
-        proc_dict = self.insert_storage_regcyclev3(proc_dict, reg_storage_idx, reg_param)
+        obj.insert_storage_regcyclev3(reg_storage_idx, reg_param)
 
         # Check that the upper charge voltage is set the same for both charge current steps
         reg_charge_idx = 27 + 1
-        assert proc_dict['MaccorTestProcedure']['ProcSteps']['TestStep'][reg_charge_idx
-               ]['Ends']['EndEntry'][1]['Value'] \
-               == proc_dict['MaccorTestProcedure']['ProcSteps']['TestStep'][reg_charge_idx + 1
+        assert obj['MaccorTestProcedure']['ProcSteps']['TestStep'][reg_charge_idx][
+                   'Ends']['EndEntry'][1]['Value'] \
+            == obj['MaccorTestProcedure']['ProcSteps']['TestStep'][reg_charge_idx + 1
                ]['Ends']['EndEntry'][0]['Value']
 
         reg_charge_idx = 59 + 1
-        assert proc_dict['MaccorTestProcedure']['ProcSteps']['TestStep'][reg_charge_idx
+        assert obj['MaccorTestProcedure']['ProcSteps']['TestStep'][reg_charge_idx
                ]['Ends']['EndEntry'][1]['Value'] \
-               == proc_dict['MaccorTestProcedure']['ProcSteps']['TestStep'][reg_charge_idx + 1
-               ]['Ends']['EndEntry'][0]['Value']
+               == obj['MaccorTestProcedure']['ProcSteps']['TestStep'][reg_charge_idx + 1
+                                                                       ]['Ends']['EndEntry'][0]['Value']
 
-        return proc_dict
+        return obj
 
-    def insert_initialrest_regcyclev3(self, proc_dict, rest_idx, index):
+    def insert_initialrest_regcyclev3(self, rest_idx, index):
         """
         Inserts initial rest into procedure dictionary at given id.
 
         Args:
-            proc_dict (dict):
             rest_idx (int):
             index (int):
 
         Returns:
             dict:
 
         """
-        steps = proc_dict['MaccorTestProcedure']['ProcSteps']['TestStep']
+        steps = self['MaccorTestProcedure']['ProcSteps']['TestStep']
         # Initial rest
         offset_seconds = 720
         assert steps[rest_idx]['StepType'] == "Rest"
         assert steps[rest_idx]['Ends']['EndEntry'][0]['EndType'] == "StepTime"
         time_s = int(round(3 * 3600 + offset_seconds * (index % 96)))
         steps[rest_idx]['Ends']['EndEntry'][0]['Value'] = time.strftime('%H:%M:%S', time.gmtime(time_s))
 
-        return proc_dict
+        return self
```

Comment:
Just want to make sure I understand what is going on here. By returning `self` its effectively updating the dictionary object in place?

Reply:
It is updating in place, by modifying self, but the return value doesn't matter, the only reason I returned self is to because some might expect the update function to return the updated object, e. g.

```
procedure = procedure.insert_thing()
```

But if we're updating in place, perhaps we shouldn't return anything, as most functions I'm aware of that return updated objects (e. g. in pandas) don't modify in-place.